compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1955,9 +1955,11 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
         isSubInfo(info1, tp2.refinedInfo.widenExpr, m.symbol.info.orElse(info1))
         || matchAbstractTypeMember(m.info)
 
-      tp1.member(name) match // inlined hasAltWith for performance
+      def memberQualifies = tp1.member(name) match // inlined hasAltWith for performance
         case mbr: SingleDenotation => qualifies(mbr)
         case mbr => mbr hasAltWith qualifies
+
+      memberQualifies || (tp1.isStable && isSub(TermRef(tp1, name), tp2.refinedInfo))
```

Comment:
Why not put it in `qualifies` instead? Then it would work for overloaded members as well.

i.e. it would be `isSubType(TermRef(tp1, n.symbol), tp2.refinedInfo)`. 

`isSub` is not used itself in TypeComparer.